bcda/auth/rsautils/keytools.go

```diff
@@ -0,0 +1,47 @@
+package rsautils
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"regexp"
+)
+
+const RSAKEYMINBITS = 2048
+
+func ReadPublicKey (publicKey string) (*rsa.PublicKey, error) {
+	if publicKey == "" {
+		return nil, fmt.Errorf("empty string provided")
+	}
+
+	emptyPEMRegex := "-----BEGIN RSA PUBLIC KEY-----(\\W*)-----END RSA PUBLIC KEY-----"
```

Comment:
Don't pem.Decode() and/or x509.ParsePKIXPublicKey() validate the structure of the key, including the header and footer? Why are we duplicating that checking?

Reply:
Good catch.  I was getting memory access violations with empty (header + footer with nothing in between) keys at one point, so I added the extra regex to filter them out, and tests to prevent regression.  But the tests pass without the extra regex filter, and without the empty string filter above it.  Useless code has now been removed.